src/types/artist.ts

```diff
@@ -18,10 +33,25 @@ export type ArtistProfile = TimeField & {
 
 export const mapArtist = (artistProfile: ArtistProfile): Artist => {
   return {
+    id: artistProfile.artistId,
     name: artistProfile.name,
+    address: controlledEthereumAddressFromId(artistProfile.artistId),
+    avatarUrl: artistProfile.avatarUrl,
     slug: slugify(`${artistProfile.name} ${artistProfile.createdAtTime.getTime()}`),
-    id: artistProfile.artistId,
     createdAtTime: artistProfile.createdAtTime,
     createdAtEthereumBlockNumber: artistProfile.createdAtEthereumBlockNumber
   }
 };
+
+export const distinctEarliestArtistProfiles = (artistProfiles: ArtistProfile[]): ArtistProfile[] => {
```

Comment:
i think we said we'd go with a simple solution that just handles updating undefined fields, rather than caring about which is earliest?

Reply:
There was unfortunately a few things that led me to just try to tackle it. While running the spindexer using the update-only solution, I realised that a batch of artist profiles sometimes contains two profiles for the same artist. 

Before running the db insert, we need to de-dup the set of artists (otherwise the "undefined-fields-only" insert statement breaks). I then found that de-duping using `uniq` or lodash's `uniqueBy` led to non deterministic choice of which artist profile is then used to seed the artist, which leads to different times, blocknumbers, and subsequently, different slugs.

So I thought that taking a similar approach to the approach we needed to take in the migration file would get us a step closer to a more reliable solution.

Reply:
hm yeah i imagine it may make sense to just accept the non-determinism to make things easier, but on some fields (eg: slugs) it could be a significant issue

Reply:
this logic still doesn't solve it fully though, for example:

 - aidan runs the indexer as is, indexes snoop dog from soundxyz, then aidan merges in a pull request that includes a new platform with some of snoop dogs much older nfts from a past platform. aidan gets the soundxyz slug
 - gian runs the indexer anew, after the above PR is merged. gian gets that oldplatform slug

even more so, I could imagine that even with the same PR, it's possible that 2 different people running a node get a different batch of nfts in their processPlatformTracks loop, one which includes snoop_sound and snoop_oldplatform an snoop_sound

so the nondeterminism emerges anyway

Reply:
we could have a distinctEarliestArtistProfiles kind of logic applied at the database upsert level (like, with a CrdtUpsert) - may be worth considering if we can utilize that since it also would make use of functionality we already have

but even so, we're still left with the issue that a slug may change. maybe that's something we should just accept

Reply:
From our chats, we're going to accept the problem with slugs changing or diverging. Last thing to do here is remove this extra logic. And also move the insert logic into upsert, along with the de-duplication.